frontend/frontend/protocols/ssh/_ssh_server.py

```diff
@@ -0,0 +1,157 @@
+import logging
+import datetime
+from typing import List, Optional, Set, Tuple
+
+from paramiko.common import (AUTH_FAILED, AUTH_SUCCESSFUL, OPEN_SUCCEEDED)
+import paramiko
+from paramiko.channel import Channel
+
+from frontend.honeylogger import SSHSession
+from ._proxy_handler import ProxyHandler
+
+
+logger = logging.getLogger(__name__)
+
+
+class Server(paramiko.ServerInterface):
```

Comment:
This class implements the interface given by paramiko. It is here we can collect data from the SSH protocol.

It currently proxies over shell requests, exec requests, and window change requests over to the proxy_handler. The result from this proxying is then relayed back here and sent back to the attacker